packages/web5/src/dwn-api.ts

```diff
@@ -0,0 +1,169 @@
+import type { Web5Agent } from '@tbd54566975/web5-agent';
+import type {
+  MessageReply,
+  RecordsReadOptions,
+  RecordsQueryOptions,
+  RecordsWriteMessage,
+  RecordsWriteOptions,
+  RecordsDeleteOptions,
+  ProtocolsQueryOptions,
+  ProtocolsConfigureOptions,
+} from '@tbd54566975/dwn-sdk-js';
+
+import { DwnInterfaceName, DwnMethodName, DataStream } from '@tbd54566975/dwn-sdk-js';
+
+import { Record } from './record.js';
+import { dataToBytes } from './utils.js';
+
+export type RecordsWriteRequest = {
+  author: string;
+  data: unknown;
+  message?: Omit<RecordsWriteOptions, 'authorizationSignatureInput' | 'data'>;
+}
+
+export type RecordsWriteResponse = MessageReply & {
+  record?: Record
+};
+
+export type RecordsQueryRequest = {
+  author: string;
+  message: Omit<RecordsQueryOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsQueryResponse = MessageReply & {
+  records: Record[]
+};
+
+export type RecordsDeleteRequest = {
+  author: string;
+  message: Omit<RecordsDeleteOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsDeleteResponse = MessageReply;
+
+export type RecordsReadRequest = {
+  author: string;
+  message: Omit<RecordsReadOptions, 'authorizationSignatureInput'>;
+}
+
+export type RecordsReadResponse = MessageReply;
+
+export type ProtocolsConfigureRequest = {
+  author: string;
+  message: Omit<ProtocolsConfigureOptions, 'authorizationSignatureInput'>;
+}
+
+export type ProtocolsQueryRequest = {
+  author: string;
+  message: Omit<ProtocolsQueryOptions, 'authorizationSignatureInput'>
+}
+
+export const DwnApi = (web5Agent: Web5Agent) => ({
+  protocols: {
+    async configure(target: string, request: ProtocolsConfigureRequest) {
+      return await web5Agent.processDwnRequest({
+        target         : target,
+        author         : request.author,
+        messageOptions : request.message,
+        messageType    : DwnInterfaceName.Protocols + DwnMethodName.Configure
+      });
+    },
+    async query(target: string, request: ProtocolsQueryRequest) {
+      return await web5Agent.processDwnRequest({
+        target         : target,
+        author         : request.author,
+        messageOptions : request.message,
+        messageType    : DwnInterfaceName.Protocols + DwnMethodName.Query
+      });
+    }
+  },
+  records: {
+    async write(target: string, request: RecordsWriteRequest): Promise<RecordsWriteResponse> {
+      const { author, data, message: requestMessage } = request;
```

Comment:
Purely a preferential thing, but IMO deconstruction induces a loss of intent. Any time I see a deconstruction, I become skeptical. 

For example, notice how you have to do `requestMessage` and `responseMessage` rather than `request.message` and `response.message`. You have to do this because you have to differentiate whether the `message` is part of the inbound or outbound. Now consider `author` and `data` -- both of which are part of the inbound but purely by looking at the variables that's unclear (I have to trace up the code to determine the context in which those things exist/originated-from).

It's one of those nitpicks, but I think it's worth nitpicking. Consider steel-manning the counter argument, which would go something like, "well `request.data` is verbose, and if I'm using `data` all over the place then I'm going to have a bunch of verbose code." While that's a good point, to that I would respond, "then your function is too large and you may need to introduce more levels of abstraction." That's not the case here, because the function is small enough that verbosity is beneficial for the purposes of clearly conveying intent.

Reply:
> Purely a preferential thing, but IMO deconstruction induces a loss of intent. Any time I see a deconstruction, I become skeptical.
> 
> For example, notice how you have to do `requestMessage` and `responseMessage` rather than `request.message` and `response.message`. You have to do this because you have to differentiate whether the `message` is part of the inbound or outbound. Now consider `author` and `data` -- both of which are part of the inbound but purely by looking at the variables that's unclear (I have to trace up the code to determine the context in which those things exist/originated-from).

such a good point.